test/functional/interface_rpc.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Tests some generic aspects of the RPC interface."""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal, str_to_b64str
+
+import http.client
+import json
+import urllib.parse
+
+class RPCInterfaceTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.setup_clean_chain = True
+
+    def send_request(self, request):
+        self.conn.request('POST', '/', json.dumps(request), self.headers)
+        out = self.conn.getresponse().read()
+        return json.loads(out.decode('ascii'))
+
+    def test_batch_request(self):
+        self.log.info("Testing basic JSON-RPC batch request...")
+
+        request = [
+            # Two basic requests that will work fine.
+            {"method": "getblockcount", "id": 1},
+            {"method": "getbestblockhash", "id": 2},
+            # Request that will fail.  The whole batch request should still
+            # work fine.
+            {"method": "invalidmethod", "id": 3},
+        ]
+        results = self.send_request(request)
```

Comment:
you're not using the `.batch()` functionality of the RPC proxy?

Reply:
No, as I didn't know it exists.  Will update the code later.

Reply:
Changed - that's indeed much simpler!